guides/common/modules/proc_upgrading-a-connected-project-server.adoc

```diff
@@ -1,45 +1,33 @@
 [id="upgrading_a_connected_{project-context}_server_{context}"]
 = Upgrading a connected {ProjectServer}
 
-Use this procedure for a {ProjectServer} with access to the public internet
+Upgrading a {ProjectServer} with access to the public internet includes backing up the server, ensuring that the system and {Project} configuration is up-to-date, and running the upgrade.
 
 include::snip_warning-maintain-config-noop.adoc[]
 
-include::snip_prerequisites-upgrading-project-server.adoc[]
-
-.Upgrade {ProjectServer}
-
-. Stop all {Project} services:
+.Procedure
+. Create a backup of your {ProjectServer}:
+.. Stop all {Project} services:
 +
 [options="nowrap" subs="+quotes,verbatim,attributes"]
 ----
 # {foreman-maintain} service stop
 ----
-. Take a snapshot or create a backup:
+.. Take a snapshot or create a backup:
 * On a virtual machine, take a snapshot.
 * On a physical machine, create a backup.
 +
-. Start all {Project} services:
+.. Start all {Project} services:
 +
 [options="nowrap" subs="+quotes,verbatim,attributes"]
 ----
 # {foreman-maintain} service start
 ----
-
-. Optional: If you made manual edits to DNS or DHCP configuration in the `/etc/zones.conf` or `/etc/dhcp/dhcpd.conf` files, back up the configuration files because the installer only supports one domain or subnet, and therefore restoring changes from these backups might be required.
-
-. Optional: If you made manual edits to DNS or DHCP configuration files and do not want to overwrite the changes, enter the following command:
-+
-[options="nowrap" subs="attributes"]
-----
-# {foreman-installer} \
---foreman-proxy-dhcp-managed=false \
---foreman-proxy-dns-managed=false
-----
-. In the {ProjectWebUI}, navigate to *Hosts* > *Discovered hosts*.
-On the Discovered Hosts page, power off and then delete the discovered hosts.
-From the *Select an Organization* menu, select each organization in turn and repeat the process to power off and delete the discovered hosts.
-Make a note to reboot these hosts when the upgrade is complete.
+. Delete discovered but unregistered hosts from {Project}.
```

Comment:
I wonder if that's really still required.

@stejskalleos do you maybe know why we tell users to delete discovered but unregistered hosts before an upgrade?

Reply:
I've never heard of this practice; it's new to me. With upgraded Foreman, you should also get upgraded Discovery, so what could be the reason for this action?

One thing could be facts - we might be adding/removing new facts necessary for the process.

Since this didn't happen for a while, it's safe to remove this step.

Reply:
It always surprised me and I never figured out why. It was added in 84e812e132603f537e539e3d61396bc55697f5dd which is basically the "take everything from downstream" commit. Then I didn't do any downstream archeology on why it's there but instead focused on other things.

Reply:
Yeah, the only thing I can think of is incompatibility of the data.

It was added for https://bugzilla.redhat.com/show_bug.cgi?id=1245390, which - once again - doesn't explain *why* it was done :roll_eyes: 

Maybe @lzap remembers what he did 10 summers ago?

Reply:
Hey, is like a miracle but I remember.

Discovery process has two paths, one is when a host is new and one is when a host is being updated. After upgrade, the (updated) path that would possibly fix something for a host (like a discovered organization or other settings that are applied) would not be executed at all. At least that was the idea.

Cheers!

Reply:
> But is it dangerous in any way if you'd skip that?

Well if you put it that way, no, not dangerous. We just happened to have a bug once that needed a fresh re-discovery so I worked with docs team to make these statements.

@aneta-petrova Feel free to drop'em if needed.

Reply:
Thanks so much @lzap! This is an important improvement towards simplifying the whole upgrading procedure. I'm dropping the steps related to deleting discovered hosts based on "We just happened to have a bug once [...]" :)